src/main/java/com/smartcar/sdk/AuthClient.java

```diff
@@ -311,6 +314,53 @@ public AuthVehicleInfo build() {
   }
 
   /**
+   * A class that creates a custom AuthUrlBuilder object, used
+   * for generating authentication URLs.
+   */
+  public class AuthUrlBuilder {
+    HttpUrl.Builder urlBuilder;
+
+    public AuthUrlBuilder() {
+      this.urlBuilder = HttpUrl.parse(AuthClient.this.urlAuthorize).newBuilder()
+              .addQueryParameter("response_type", "code")
+              .addQueryParameter("client_id", AuthClient.this.clientId)
+              .addQueryParameter("redirect_uri", AuthClient.this.redirectUri)
+              .addQueryParameter("mode", AuthClient.this.testMode ? "test" : "live");
+      if (AuthClient.this.scope.length != 0) {
+        this.urlBuilder.addQueryParameter("scope", Utils.join(AuthClient.this.scope, " "));
+      }
+    }
+
+    public AuthUrlBuilder setState(String state) {
+      if (state != "") {
+        urlBuilder.addQueryParameter("state", state);
+      }
+      return this;
+    }
+
+    public AuthUrlBuilder setApprovalPrompt(boolean approvalPrompt) {
+      urlBuilder.addQueryParameter("approval_prompt", approvalPrompt ? "force" : "auto");
+      return this;
+    }
+
+    public AuthUrlBuilder setSingleSelect(boolean singleSelect) {
+      urlBuilder.addQueryParameter("single_select", Boolean.toString(singleSelect));
+      return this;
+    }
+
+    public AuthUrlBuilder setMakeBypass(String make) {
+      urlBuilder.addQueryParameter("make", make);
+      return this;
+    }
+
+    public String build() {
+      return this.urlBuilder.build().toString();
```

Comment:
Why use `this.urlBuilder` here and not just `urlBuilder`?

Reply:
good catch!